Core/Layer/Worlds/StatusBar/StatusBarRenderer.cs

```diff
@@ -0,0 +1,1160 @@
+            }
+        }
+
+        List<(string Patch, int Width, int Offset)> glyphs = new(text.Length);
+
+        foreach (char c in text)
+        {
+            string patch = GetFontPatch(hud, fontDef, c);
+            int width;
+            int xOffset = 0;
+
+            if (hud.Textures.TryGet(patch, out var handle))
+                width = handle.Dimension.Width;
+            else
+            {
+                 if (hud.Textures.TryGet(patch, out handle, ResourceNamespace.Sprites))
+                     width = handle.Dimension.Width;
+                 else
+                     continue;
+            }
+            
+            if (fontDef.Type == 0 || fontDef.Type == 1)
+            {
+                if (monoWidth > 0)
+                {
+                    if (width < monoWidth)
+                        xOffset = (monoWidth - width) / 2;
+                    width = monoWidth;
+                }
+            }
+
+            glyphs.Add((patch, width, xOffset));
+            totalWidth += width;
+        }
+
+        int drawX = pos.X;
+        int drawY = pos.Y;
+
+        if ((number.Alignment & StatusBarAlignment.HCenter) != 0)
+            drawX -= totalWidth / 2;
+        else if ((number.Alignment & StatusBarAlignment.Right) != 0)
+            drawX -= totalWidth;
+
+        Align yAnchor = Align.TopLeft;
+        if ((number.Alignment & StatusBarAlignment.Bottom) != 0) yAnchor = Align.BottomLeft;
+        
+        bool isBottomAnchor = yAnchor == Align.BottomLeft;
+
+        foreach (var (patch, width, xOffset) in glyphs)
+        {
+            Vec2I drawPos = (drawX + xOffset, drawY);
+            
+            if (isBottomAnchor) drawPos.Y -= 1;
+
+            if (hud.Textures.HasImage(patch))
+                hud.Image(patch, drawPos, anchor: yAnchor, color: drawColor, alpha: alpha);
+            else if (hud.Textures.HasImage(patch, ResourceNamespace.Sprites))
+                hud.Image(patch, drawPos, anchor: yAnchor, resourceNamespace: ResourceNamespace.Sprites, color: drawColor, alpha: alpha);
+            
+            drawX += width;
+        }
+
+        DrawChildren(hud, number, pos, containerHeight, context, widescreenOffset);
+    }
+    
+    private static void DrawStatTotals(IHudRenderContext hud, StatusBarComponentDef comp, Vec2I pos, 
+        StatusBarHudFontDef? fontDef, int fontHeight, float alpha)
+    {
+        var stats = WorldStatic.World.LevelStats;
+        
+        var statList = new[] 
+        {
+            ("K", stats.KillCount, stats.TotalMonsters),
+            ("I", stats.ItemCount, stats.TotalItems),
+            ("S", stats.SecretCount, stats.TotalSecrets)
+        };
+
+        Vec2I cursor = pos;
+        bool isVertical = comp.Vertical;
+        StatusBarAlignment alignment = comp.Alignment;
+
+        // Local helper function to avoid repeating draw/fallback/measure logic
+        int DrawTextPart(string text, Vec2I position, string? translation)
```

Comment:
Unfortunately this will have to get pulled out so it's not inside this function. .NET will allocate a new class encapsulating the variables from the parent functions that would go to the GC.